client/packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModalContext.ts

```diff
@@ -13,6 +13,7 @@ export interface ConfirmationModalState {
     | ((state: ConfirmationModalState) => void)
     | ((state: ConfirmationModalState) => Promise<void>);
   onCancel?: (() => void) | (() => Promise<void>);
+  cleanupConfirm?: () => void;
```

Comment:
Vax modal can now have 2 confirmation modals (confirm early vaccination, confirm no stock line selected). 

Confirming the first modal triggers the second, but existing confirmation modal code calls setOpen(false) _after_ calling onConfirm (so, onConfirm opens the second modal, but then the setOpen(false) closes it)

So, added a cleanup override, to allow us to close the modal _before_ calling onConfirm in this specific case